darts/tests/models/forecasting/test_ensemble_models.py

```diff
@@ -45,17 +45,55 @@ def test_untrained_models(self):
         model = NaiveDrift()
         _ = NaiveEnsembleModel([model])
 
-        # trained models should raise error
+        # trained local models should raise error
         model.fit(self.series1)
         with self.assertRaises(ValueError):
             NaiveEnsembleModel([model])
 
         # an untrained ensemble model should also give untrained underlying models
         model_ens = NaiveEnsembleModel([NaiveDrift()])
         model_ens.fit(self.series1)
-        assert model_ens.models[0]._fit_called
+        assert model_ens.forecasting_models[0]._fit_called
         new_model = model_ens.untrained_model()
-        assert not new_model.models[0]._fit_called
+        assert not new_model.forecasting_models[0]._fit_called
+
+    def test_trained_models(self):
+        """EnsembleModels can be instantiated with pre-trained GlobalForecastingModels"""
+        local_model = NaiveDrift()
+        global_model = LinearRegressionModel(lags=2)
+        local_model.fit(self.series1)
+        global_model.fit(self.series1)
+
+        # local and global trained
+        with self.assertRaises(ValueError):
+            NaiveEnsembleModel([local_model, global_model])
+
+        # local untrained, global trained
+        with self.assertRaises(ValueError):
+            NaiveEnsembleModel([local_model.untrained_model(), global_model])
+
+        # local trained, global untrained
+        with self.assertRaises(ValueError):
+            NaiveEnsembleModel([local_model, global_model.untrained_model()])
+
+        # global trained, global untrained
+        with self.assertRaises(ValueError):
+            NaiveEnsembleModel([global_model, global_model.untrained_model()])
+
+        # both global trained, retrain = True
+        model_ens_retrain = NaiveEnsembleModel(
+            [global_model, global_model], retrain_forecasting_models=True
```

Comment:
I wonder if it was better before (I know I suggested to implement what you have now :D ) when we raised an error instead of silently calling model.untrained_model().
Let me know what you think.

Reply:
I prefer to raise error, to make it as explicit/transparent as possible